modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPAttachmentFileEntryLocalServiceImpl.java

```diff
@@ -257,6 +257,13 @@ else if (Validator.isNotNull(externalReferenceCode)) {
 					externalReferenceCode, serviceContext.getCompanyId());
 		}
 
+		if ((cpAttachmentFileEntry != null) &&
+			(cpAttachmentFileEntry.getClassPK() != classPK)) {
+
+			cpAttachmentFileEntry = null;
+			externalReferenceCode = null;
+		}
```

Comment:
```suggestion
		if (cpAttachmentFileEntry.getClassPK() != classPK) {

throw new DuplicateCPAttachmentFileEntry		}
```

and move this block inside the 

else if (Validator.isNotNull(externalReferenceCode)) 

block

Reply:
but then it will only throw the duplicate exception if the user sends a duplicated ERC. What about if the user sends a duplicated Attachment Id? Then the same issue will happen, the existing attachment will be fetched and updated even if it does not belong to the correct product.